vpr/src/base/partition.h

```diff
@@ -0,0 +1,27 @@
+#ifndef PARTITION_H
+#define PARTITION_H
+
+#include "vtr_strong_id.h"
+#include "region.h"
+
+/**
+ * This class stores the data for each constraint partition -
```

Comment:
Give more context.  
This file defines the data for a partition: a grouping of atoms that are constrained to some portion of an FPGA. A partition defines the atoms that are assigned to it, and the locations in which they can be placed.  The locations are be a union of rectangles in x,y,sub_tile space. Common cases include a single rectangle or a single x,y,sub_tile (specific location); more complex partitions with L, T or other shapes can be created with a union of multiple rectangles.